server/catgenome/src/main/java/com/epam/catgenome/util/SimpleMotifSearchIterator.java

```diff
@@ -0,0 +1,179 @@
+        this.sequence = seq;
+        this.offset = start;
+        this.includeSequence = includeSequence;
+
+        final String iupacRegex = MotifSearcher.convertIupacToRegex(regex);
+        final String invertedRegex = invertCurrentRegex(regex.toUpperCase(Locale.ROOT));
+        final Pattern patternPositive = Pattern.compile(iupacRegex, Pattern.CASE_INSENSITIVE);
+        final Pattern patternNegative = Pattern.compile(invertedRegex, Pattern.CASE_INSENSITIVE);
+        if (strand == null) {
+            this.positiveMatches = populatePositiveMatches(patternPositive.matcher(new String(seq)));
+            this.negativeMatches = populatePositiveMatches(patternNegative.matcher(new String(seq)));
+        } else if (strand == StrandSerializable.POSITIVE) {
+            this.positiveMatches = populatePositiveMatches(patternPositive.matcher(new String(seq)));
+            this.negativeMatches = new LinkedList<>();
+        } else {
+            this.positiveMatches = new LinkedList<>();
+            this.negativeMatches = populatePositiveMatches(patternNegative.matcher(new String(seq)));
+        }
+    }
+
+    @Override
+    public boolean hasNext() {
+        return !positiveMatches.isEmpty() || !negativeMatches.isEmpty();
+    }
+
+    @Override
+    public Motif next() {
+        Match match;
+        StrandSerializable currentStrand;
+        if (negativeMatches.isEmpty()) {
+            match = positiveMatches.removeFirst();
+            currentStrand = StrandSerializable.POSITIVE;
+        } else if (positiveMatches.isEmpty() || negativeMatches.peekLast().start < positiveMatches.peekFirst().start) {
+            match = negativeMatches.removeLast();
+            currentStrand = StrandSerializable.NEGATIVE;
+        } else {
+            match = positiveMatches.removeFirst();
+            currentStrand = StrandSerializable.POSITIVE;
+        }
+        if (includeSequence) {
+            return getMotif(contig, match.start, match.end, currentStrand);
+        }
+        return new Motif(contig, match.start + offset, match.end + offset, currentStrand, null);
+    }
+
+    private String invertCurrentRegex(final String regex) {
+        final byte[] chars = regex.getBytes();
+        for (int i = 0; i < chars.length; i++) {
+            chars[i] = checkAndRevert(chars[i]);
+        }
+        String invertedRegex = new StringBuilder(new String(chars)).reverse().toString();
+        final Map<String, String> links = MotifSearcher.getLinks();
+        final Map<String, String> codes = MotifSearcher.getCodes();
+        for (Map.Entry<String, String> entry : links.entrySet()) {
+            if (invertedRegex.contains(entry.getKey())) {
+                invertedRegex = invertedRegex.replaceAll(entry.getKey(), codes.get(entry.getValue()));
+            }
+        }
+        return invertedRegex.toLowerCase(Locale.ROOT);
+    }
+
+    private byte checkAndRevert(final byte value) {
+        final byte openBoxBrace = '[';
```

Comment:
all these vars should be constants, we don't need to create it every time we use this method